bin/varnishd/http1/cache_http1_deliver.c

```diff
@@ -163,11 +165,11 @@ V1D_Deliver(struct req *req, struct boc *boc, int sendbody)
 
 	if (!sendbody) {
 		AZ(req->wrk->v1l);
-		VDP_close(req);
```

Comment:
so where is this happening?

Reply:
The VDP_close() call was moved up one stack frame to cache_req_fsm.c, instead of relying on the individual transports to do it before returning. This makes it easier to know that it will always be called, and at the same time elliminating memory leaks from failure to call it for each of the H/1 canned responses in reaction to WS exhaustion (they failed to call VDP_close()).